pkg/prometheus/operator.go

```diff
@@ -229,6 +229,16 @@ func (sr *StatusReporter) Process(ctx context.Context, p monitoringv1.Prometheus
 		}
 	}
 
+	// Indicate that no scrape resource selectors are specified. Do not take precedence over an unhealthy condition.
+	allSelectorsNil := commonFields.ServiceMonitorSelector == nil &&
```

Comment:
since we have the same test over a few places, a utility function might be handy.